File: frontend/react/src/types/outreachDraft.ts
// Mirrors backend/schemas/outreach_draft.py's OutreachDraftOut exactly.
// `status` is always "Draft" at creation; "Approved"/"Archived" via the
// two human-reviewer actions, "Sent" only via a real send through
// POST /{id}/send (outreach workflow redesign - generation itself never
// changes status).

import type { GroundedInItem } from "./groundedIn";

export interface OutreachDraft {
  id: string;
  company_id: string;
  opportunity_id: string | null;
  type: string;
  subject: string | null;
  content: string;
  status: string;
  created_at: string;
  // Only present on the detail response, not in list results (V3
  // Enhancements Phase 3B).
  grounded_in?: GroundedInItem[];
}

File: backend/schemas/outreach_draft.py
"""Response schema for the Outreach Draft read endpoints (V3 Phase 7C).
Mirrors backend/database/models/outreach_draft.py's OutreachDraft
exactly. `status` reflects the repository's existing invariant - always
"Draft" at creation, "Approved"/"Archived" only via a human reviewer's
explicit action (mark_draft_approved/mark_draft_archived); this schema
does not change or weaken that invariant, only displays its value.
"""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class OutreachDraftOut(BaseModel):
    id: str
    company_id: str
    opportunity_id: Optional[str] = None
    type: str
    subject: Optional[str] = None
    content: str
    status: str
    created_at: datetime

    model_config = {"from_attributes": True}
